arch/x86/common/source/assert.cpp

```diff
@@ -81,7 +81,7 @@ __attribute__((noreturn)) void pre_new_sweb_assert(const char* condition, uint32
   if (ArchMulticore::numRunningCPUs() > 1)
   {
       ArchMulticore::stopAllCpus();
-      volatile size_t wait = 0x10000000;
+      eastl::atomic<size_t> wait = 0x10000000;
```

Comment:
I'm not sure this is correct. indeed volatile is deprecated for many use cases where users commonly had the misconception that volatile would provide some form of thread safety... but here we literally want what volatile is there for: "do not optimize out any operations on this variable".

do i misremember that atomic<> may be optimized out?

Reply:
That line no longer exists in the current version of the file. 
https://github.com/Woazboat/sweb/blob/multicore/arch/x86/common/source/assert.cpp#L98

The reason why it was there in the first place was as a quick interim measure to prevent backtraces and assert messages from individual cpu cores from overlapping and interspersing, but that has shortly afterwards been replaced by proper locking/serialization.